tests/callistocore/models.py

```diff
@@ -0,0 +1,124 @@
+import hashlib
+
+import nacl.secret
+import nacl.utils
+
+from django.conf import settings
+from django.utils.crypto import get_random_string, pbkdf2
+
+
+def _encrypt_report(salt, key, report_text):
+    """Encrypts a report using the given secret key & salt. The secret key is stretched to 32 bytes using Django's
+    PBKDF2+SHA256 implementation. The encryption uses PyNacl & Salsa20 stream cipher.
+
+    Args:
+      salt (str): cryptographic salt
+      key (str): secret key
+      report_text (str): full report as a string
+
+    Returns:
+      bytes: the encrypted bytes of the report
+
+    """
+    stretched_key = pbkdf2(key, salt, settings.KEY_ITERATIONS, digest=hashlib.sha256)
+    box = nacl.secret.SecretBox(stretched_key)
```

Comment:
might as well call the non-test models version of _encrypt_report after stretching the key